src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -2211,6 +2211,20 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                 break;
             }
 
+            case NI_Sve_CreateBreakAfterMask: 
+            case NI_Sve_CreateBreakBeforeMask:
+            {
+                GetEmitter()->emitInsSve_R_R_R(ins, emitSize, targetReg, op1Reg, op2Reg, INS_OPTS_SCALABLE_B);
```

Comment:
For [BRKB](https://docsmirror.github.io/A64/2023-06/brkb_p_p_p.html) and [BRKA](https://docsmirror.github.io/A64/2023-06/brka_p_p_p.html), if we do not pass `insScalableOpts`, it is treated as `BRK* Pd.B, Pg/Z, Pn.B`. How do we generate the merge variant? If we want to support one of those, should merge be the default one? Accordingly we should also update the summary docs and remove the instruction that we are not supporting. We also need to think about what if all `targetReg`, `op1Reg` and `op2Reg` end up to same register or two of them have same register, does it change the operation of `Pg/M` vs. `Pg/Z`. Lastly, we should add test coverage (if not there currently) where we pass same value to `mask` and `srcMask` to simulate such behavior.

Reply:
When looking at the original generated API, there were two C intrinsic variants that it listed:
M variant https://dougallj.github.io/asil/doc/brka_p_p_p_z_8.html
Z variant https://dougallj.github.io/asil/doc/brka_p_p_p_z_8.html

Based on the API signature of our intrinsic, it matches exactly to the Z variant  - so naturally I think that's most likely the correct one. I figured the M variant that was listed is an artifact of the API generation; as other APIs have before. This also leads me to believe that exposing each variant needs to have their own API.

Reply:
> I figured the M variant that was listed is an artifact of the API generation

Yes, the API generation will have just blindly put them together.

> This also leads me to believe that exposing each variant needs to have their own API.

We can we replicate the zero version using:
```
CndSel(mask, CreateBreakAfterMask(mask, op1), zero)
```

And then in lowering or somewhere, optimise that to a single BRKA with zeroing.

(Or, if I'm wrong and that doesn't work, then yes, ideally add a extra API)

Doesn't need doing now, but could you raise a ticket please.

Reply:
> We can we replicate the zero version using:

Do you mean "merge"? The API is the zero version currently

Reply:
> could you raise a ticket please.

+1 on that if we decide to skip a variant of the instruction.

Reply:
I agree with just keeping "Z" version. Just update the summary docs.

https://github.com/dotnet/runtime/pull/104184/files#diff-65452cc3b51792a27a23d428011052b18c0b7de4576424a195e0cf5003dc6a9cR987-R988